php_laravel/app/Http/Controllers/ChargeController.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\View\View;
+use Illuminate\Http\RedirectResponse;
+use OmiseCharge;
+use OmiseSource;
+use OmiseToken;
+use Illuminate\Support\Facades\Http;
+use App\Models\Payment;
+
+class ChargeController extends Controller
+{
+    //
+    public function index() : View {
+        return view('formcharge');
+    }
+
+    /**
+     * 基本的にchargeをするときは、tokenを作って、それを元にchargeを作るものと理解してる
+     */
+    public function cardCharge(Request $req) : View {
+
+        // print_r($req->name);
+        \Log::debug($req->name);
+
+        $token = OmiseToken::create(array(
+            'card' => array(
+              'name' => $req->name,
+              'number' => trim($req->card_number),
+              'expiration_month' => $req->expired_month,
+              'expiration_year' => $req->expired_year,
+              'security_code' => $req->security_code,
+            )
+          ));
+        
+
+        $charge = OmiseCharge::create(array(
+            'amount' => $req->money,
+            'currency' => 'jpy',
+            'card' => $token['id']
+        ));
+
+        // print_r($charge);
+
+        return view('charge-result', ['charge'=>$charge]);
+    }
+
+    /**
+     * 
+     */
+    public function cardChargeOmise(Request $req) : View {
+
+        \Log::debug($req);
+
+        $charge = OmiseCharge::create(array(
+            'amount' => 100,
+            'currency' => 'jpy',
+            'card' => $req->omiseToken
+        ));
+
+        // print_r($charge);
+
+        return view('charge-result', ['charge'=>$charge]);
+    }
+
+    /**
+     * 
+     */
+    public function cardChargeOmiseAnother(Request $req) : View {
+
+        \Log::debug($req);
+
+        $charge = OmiseCharge::create(array(
+            'amount' => $req->money,
+            'currency' => 'jpy',
+            'card' => $req->omiseToken
+        ));
+
+        // print_r($charge);
+
+        return view('charge-result', ['charge'=>$charge]);
+    }
+
+    private function random($length = 8)
+    {
+        return substr(str_shuffle('1234567890abcdefghijklmnopqrstuvwxyz'), 0, $length);
+    }
+
+    /**
+     * 基本的にchargeをするときは、tokenを作って、それを元にchargeを作るものと理解してる
+     */
+    public function paypayCharge(Request $req) : RedirectResponse {
+
+        $apiURL = 'https://api.omise.co/charges';
+
+        $encoded = base64_encode(env('OMISE_SECRET_KEY') . ':' );
```

Comment:
I see we added `config/omise.php`. So, it would be better to have an example that fetches the key from a config file rather than directly from env.

Reply:
Ok, will change using config file.

Reply:
@aashishgurung 
Done
https://github.com/omise/examples/pull/29/commits/64d875dae2316e04d267dd2750d5c4c4f75ceae5